Issue #11514: Improve code coverage by adding a new ParseTreeTableModelTest

alimhtsai opened the pull request:
Solves issue #11514 Improve code coverage for GUI classes

- Built successfully with `mvn clean verify`
- Added a new ParseTreeTableModelTest
- Increased the code coverage of ParseTreeTableModel class

```
$ java -jar checkstyle-10.9.2-all.jar -T InputParseTreeTablePresentation.java 
COMPILATION_UNIT -> COMPILATION_UNIT [1:0]
|--PACKAGE_DEF -> package [1:0]
|   |--ANNOTATIONS -> ANNOTATIONS [1:43]
|   |--DOT -> . [1:43]
|   |   |--DOT -> . [1:39]
|   |   |   |--DOT -> . [1:28]
|   |   |   |   |--DOT -> . [1:22]
|   |   |   |   |   |--DOT -> . [1:11]
|   |   |   |   |   |   |--IDENT -> com [1:8]
|   |   |   |   |   |   `--IDENT -> puppycrawl [1:12]
|   |   |   |   |   `--IDENT -> tools [1:23]
|   |   |   |   `--IDENT -> checkstyle [1:29]
|   |   |   `--IDENT -> gui [1:40]
|   |   `--IDENT -> parsetreetablepresentation [1:44]
|   `--SEMI -> ; [1:70]
`--CLASS_DEF -> CLASS_DEF [6:0]
    |--MODIFIERS -> MODIFIERS [6:0]
    |--BLOCK_COMMENT_BEGIN -> /* [3:0]
    |   |--COMMENT_CONTENT -> *\n* class javadoc\n [3:2]
    |   `--BLOCK_COMMENT_END -> */ [5:0]
    |--LITERAL_CLASS -> class [6:0]
    |--IDENT -> InputParseTreeTablePresentation [6:6]
    `--OBJBLOCK -> OBJBLOCK [6:38]
        |--LCURLY -> { [6:38]
        |--VARIABLE_DEF -> VARIABLE_DEF [9:4]
        |   |--MODIFIERS -> MODIFIERS [9:4]
        |   |--TYPE -> TYPE [9:4]
        |   |   |--BLOCK_COMMENT_BEGIN -> /* [8:4]
        |   |   |   |--COMMENT_CONTENT -> * attribute javadoc [8:6]
        |   |   |   `--BLOCK_COMMENT_END -> */ [8:24]
        |   |   `--LITERAL_INT -> int [9:4]
        |   |--IDENT -> attribute [9:8]
        |   `--SEMI -> ; [9:17]
        |--METHOD_DEF -> METHOD_DEF [14:4]
        |   |--MODIFIERS -> MODIFIERS [14:4]
        |   |   |--BLOCK_COMMENT_BEGIN -> /* [11:4]
        |   |   |   |--COMMENT_CONTENT -> *\n    * method javadoc\n     [11:6]
        |   |   |   `--BLOCK_COMMENT_END -> */ [13:4]
        |   |   `--LITERAL_PUBLIC -> public [14:4]
        |   |--TYPE -> TYPE [14:11]
        |   |   `--LITERAL_VOID -> void [14:11]
        |   |--IDENT -> method [14:16]
        |   |--LPAREN -> ( [14:22]
        |   |--PARAMETERS -> PARAMETERS [14:23]
        |   |--RPAREN -> ) [14:23]
        |   `--SLIST -> { [14:25]
        |       |--BLOCK_COMMENT_BEGIN -> /* [15:8]
        |       |   |--COMMENT_CONTENT ->  just comment  [15:10]
        |       |   `--BLOCK_COMMENT_END -> */ [15:23]
        |       `--RCURLY -> } [16:4]
        `--RCURLY -> } [18:0]
```

romani commented:
thanks a lot !!
can you remove 
https://github.com/checkstyle/checkstyle/blob/70946a8f01ae95c5debf3f7ea10557daadea1cd6/pom.xml#L834-L849

to make it clear that coverage improved.

romani commented on src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTableModelTest.java:
```diff
@@ -0,0 +1,402 @@
+    @BeforeEach
+    public void loadTree() throws Exception {
+        tree = JavaParser.parseFile(new File(getPath("InputParseTreeTablePresentation.java")),
+            JavaParser.Options.WITH_COMMENTS).getFirstChild().getNextSibling();
```
> getFirstChild().getNextSibling();

please name field as node that you try to make as main.

romani replied:
```
$ java -jar checkstyle-10.9.2-all.jar -T InputParseTreeTablePresentation.java 
COMPILATION_UNIT -> COMPILATION_UNIT [1:0]
|--PACKAGE_DEF -> package [1:0]
|   |--ANNOTATIONS -> ANNOTATIONS [1:43]
|   |--DOT -> . [1:43]
|   |   |--DOT -> . [1:39]
|   |   |   |--DOT -> . [1:28]
|   |   |   |   |--DOT -> . [1:22]
|   |   |   |   |   |--DOT -> . [1:11]
|   |   |   |   |   |   |--IDENT -> com [1:8]
|   |   |   |   |   |   `--IDENT -> puppycrawl [1:12]
|   |   |   |   |   `--IDENT -> tools [1:23]
|   |   |   |   `--IDENT -> checkstyle [1:29]
|   |   |   `--IDENT -> gui [1:40]
|   |   `--IDENT -> parsetreetablepresentation [1:44]
|   `--SEMI -> ; [1:70]
`--CLASS_DEF -> CLASS_DEF [6:0]
....
```

did you mean `classDef` ?

romani replied:
Please rename, to make code less cryptic.

romani reviewed (changes requested):
items:

alimhtsai commented:
Hi @romani, 

This PR doesn't cover everything, I only added test cases for the ParseTreeTableModel class. 
The method coverage increases from 64% to 88%, and the line coverage increases from 71% to 91%.

Can I remove only `<exclude>com.puppycrawl.tools.checkstyle.gui.ParseTreeTableModel</exclude>` in the `pom.xml`?

Thanks!

romani commented:
just update to number got archived, to make sure nobody will not reduce % of coverage in this file in future.

romani commented on src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTableModelTest.java:
```diff
@@ -0,0 +1,402 @@
+        node = node.getFirstChild()
+            .getNextSibling()
+            .getNextSibling()
+            .getNextSibling();
```
All such magical cryptic selection of node concerns me.
As we do not test here exact tree structure, let's use util methods that we have `findFistChild(type)` or something similar, we have a bunch, please find util methods, it will ease code readability.

Please apply to all chained get calls.

romani replied:
Still Not done.

alimhtsai replied:
done

romani reviewed (changes requested):
Items

alimhtsai commented:
Hi @romani, 

I've renamed the variable as `classDef`, and updated the cryptic selection of `classDef` into util method. 
I've checked other chained get calls, and I believed that other parts are used properly. 

Thanks.

romani commented on src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTableModelTest.java:
```diff
@@ -0,0 +1,404 @@
+
+    @Test
+    public void testGetIndexOfChild() {
+        DetailAST ithChild = classDef.getFirstChild();
```
child

alimhtsai replied:
done

romani commented on src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTableModelTest.java:
```diff
@@ -0,0 +1,404 @@
+     */
+    @Test
+    public void testGetValueAt() {
+        classDef = classDef.getLastChild().getPreviousSibling();
```
Do not override classadef with other value. Make new variable and name it by what it contains

alimhtsai replied:
done

romani reviewed (changes requested):
Items

alimhtsai commented:
Hi @romani, 

I've updated the variable name and modified all other chained get calls to util method. 

Thanks.

romani commented:
@alimhtsai , please reply done for each review items I opening. I will help a lot for you to double check that nothing is forgotten and for me.

romani reviewed (changes requested):
items:

alimhtsai commented on src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTableModelTest.java:
```diff
@@ -0,0 +1,404 @@
+    @BeforeEach
+    public void loadTree() throws Exception {
+        classDef = JavaParser.parseFile(new File(getPath("InputParseTreeTablePresentation.java")),
+            JavaParser.Options.WITH_COMMENTS).getFirstChild().getNextSibling();
```
done

alimhtsai commented on src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTableModelTest.java:
```diff
@@ -0,0 +1,404 @@
+    @Test
+    public void testCommentChildCount() {
+        final DetailAST commentContentNode = classDef
+                .getFirstChild().getNextSibling().getFirstChild();
```
done

romani commented on src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTableModelTest.java:
```diff
@@ -0,0 +1,409 @@
+     *  - LITERAL_CLASS
+     *  - IDENT -> this is the classDef that holds the class name
+     *  Line number 4 - first three lines are taken by javadoc
+     *  Column 6 - first five columns taken by 'class '
```
As we make code a bit more readable by `classDef.findFirstToken(IDENT)` , do you really think vee need this comment ?

alimhtsai replied:
done, delete it

romani reviewed (changes requested):
Last item:

romani commented:
Please fix inspection CI failure

romani reviewed (approved):
Ok to merge if CI pass
